sql/parser/sql.y

```diff
@@ -572,7 +602,10 @@ alter_table_cmd:
   //     [ USING <expression> ]
 | ALTER opt_column name opt_set_data TYPE typename opt_collate_clause alter_using {}
   // ALTER TABLE <name> ADD CONSTRAINT ...
-| ADD table_constraint {}
+| ADD table_constraint
```

Comment:
Seems strange that you can add a unique index here as well as via `CREATE UNIQUE INDEX`. I wonder if we should restructure this so that you can only add non-index constraints via `ALTER TABLE`.